controllers/gitopsservice_controller.go

```diff
@@ -112,7 +112,11 @@ func (r *ReconcileGitopsService) SetupWithManager(mgr ctrl.Manager) error {
 			builder.WithPredicates(predicate.NewPredicateFuncs(func(obj client.Object) bool {
 				return obj.GetName() == "openshift-gitops"
 			})),
-		).
+		).Watches(&argoapp.ArgoCD{},
+		&handler.EnqueueRequestForObject{},
+		builder.WithPredicates(predicate.NewPredicateFuncs(func(obj client.Object) bool {
+			return obj.GetName() == "openshift-gitops"
```

Comment:
probably we can include namespace too. `obj.GetNamespace() == "openshift-gitops"`